### PR TITLE
Easy deployment of MPCAutofill using Docker

### PR DESCRIPTION
**Pull request by fklemme**
This pull-request adds Docker and Docker-Compose files for an easy setup and deployment of MPCAutofill. Besides the configuration of the `drives.csv` and the Google Service Account, most other steps are automated. Please have a look at the Readme for further information.

The following goals are aimed to be achieved in this PR (unchecked are to-do's):
- [x] Minimal software installation, requires only docker and docker-compose.
- [x] Easy setup, only individual configurations are required, such as setting up your `drives.csv`.
- [x] Decoupled, introduces only minimal changes to the original code base. All additions can be ignored if Docker is not used.
- [x] Persistent databases, enabling fast pause and resume of containers.
- [x] Chronjobs, keeping the containers up to date automatically.
- [x] Updated Readme, explaining the installation using Docker.

Feel free to test the Docker setup and propose changes. Any help or feedback is welcome.

### CODE REVIEW

**Comment by bwsinger** [comment]
> Updated Readme, hinting towards the possibility of using Docker.

Once my README PR is accepted, I'll make further edits to it and the docker README to make users aware of their options.

**Comment by ndepaola** [comment]
holy shit i love you - can't wait to test this out tomorrow!

**Comment by fklemme** [comment]
> Update client (Google Scripts) as well? Is this needed?

I tested the current state with the already available client release on this Github page and it worked just fine. Apparently, there is no interdependency between the server-side Google Service Account and the called Google Scripts in the client. Therefore, this point does not require any handling here and is removed from the TODOs.

**Comment by fklemme** [comment]
I think the PR is ready to be merged. Let me know if further changes are required.

@bwsinger, I added a small note to the README but I think it should be easy for you to merge the changes back into your PR and change them to your needs.

File: README.md [hunk 1]
```diff
@@ -28,3 +28,6 @@ Other:
 * `0 0 * * * bash /root/mpc-autofill/update_database >> /root/db_update.txt 2>&1`
 * `0 0 * * SUN bash /root/mpc-autofill/sync_dfcs`
 8. Deploy two Google Script according to the code specified in `autofill.py` and adjust the URLs in that script to point to your GS endpoints
+
+## Using Docker
```
Comment (ndepaola) [inline README.md:32]: let's move this to the top of the readme since i imagine most people will use docker to run the project

Comment (fklemme) [reply README.md]: I completely restructured the main README and merged the one from the docker sub-folder. Now, the first part is about preparative steps (setting up Google Server Account, etc.) and the second part is about the technical installation. Have a look at the rendered README at https://github.com/fklemme/mpc-autofill. I left the manual installation part (which is now at the end) to PR #33.

**Comment by ndepaola** [comment]
image

wondering if you've seen this one before?

File: README.md (hunk 1 above)
Comment (ndepaola) [reply README.md]: looks good! let's add a note for windows installation though. PR #33 might want to rebase onto this branch?

Comment (bwsinger) [reply README.md]: Yeah, I think that's the plan. Let's get this PR merged. Then I'll use bits and pieces from my verbose README to flesh out this one.

**Comment by fklemme** [comment]
> let's add a note for windows installation though.

> wondering if you've seen this one before?

Gotta admit, I didn't even know that docker/docker-compose exists for Windows. :P Let me try to install it and see if I can reproduce the error.

**Comment by fklemme** [comment]
To my very surprise, running docker-compose under Windows worked absolutely flawlessly. :D I added some instructions to the README.

Apparently, the scripts `docker/django/check_*.sh` are not found in your setup. Can you check if you have them in your repo? I added them just recently, so maybe something went wrong when you updated your repo?

On a clean checkout, the scripts should actually stop docker and yield a human-readable error message like this:

docker_unconfigured

But after installing `client_secrets.json` and `drives.csv`, docker should run just fine.

**Comment by fklemme** [comment]
@ndepaola: What does it mean for MPCAutofill if a Google Drive has `public = FALSE` in the `drives.csv`? When would I use this option?

**Comment by fklemme** [comment]
To easy. :sunglasses:

docker_demo_windows

**Comment by ndepaola** [comment]
weird that it didn't work when i checked out your branch but did work when i downloaded your repo as .zip and went from there! did i say i love you btw? this works beautifully and i can't wait for other people to try it!!

**Comment by ndepaola** [comment]
> 
> 
> @ndepaola: What does it mean for MPCAutofill if a Google Drive has `public = FALSE` in the `drives.csv`? When would I use this option?

`public=FALSE` means the drive's link won't be visible on the contributions page or search settings popup

**Comment by fklemme** [comment]
> weird that it didn't work when i checked out your branch but did work when i downloaded your repo as .zip and went from there! did i say i love you btw? this works beautifully and i can't wait for other people to try it!!

Love you for building MPCAutofill! <3 :smile:

**Pull request merged**